grakn-graql/src/main/java/ai/grakn/graql/internal/gremlin/spanningtree/KBestArborescences.java

```diff
@@ -0,0 +1,228 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016  Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn.graql.internal.gremlin.spanningtree;
+
+import ai.grakn.graql.internal.gremlin.spanningtree.graph.DirectedEdge;
+import ai.grakn.graql.internal.gremlin.spanningtree.graph.WeightedGraph;
+import ai.grakn.graql.internal.gremlin.spanningtree.util.Pair;
+import ai.grakn.graql.internal.gremlin.spanningtree.util.Weighted;
+import com.google.common.base.Optional;
```

Comment:
We're using Guava's `Optional` in a few places, but we should stick to the built-in Java one.